### PR TITLE
Tensor with BOOL type && Dynamic Slice

### PR DESCRIPTION
**Pull request by ayermolo**
Summary:
Updated tensor loading to support BOOL type.
Updated Slice to support ONNX OPSET 10. Steps not supported.
Documentation:
https://github.com/onnx/onnx/blob/master/docs/Operators.md#Slice
https://github.com/onnx/onnx/pull/1836#issuecomment-495797556
Test Plan:
Unit Tests
[Optional Fixes #issue]

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: lib/Importer/ONNXModelLoader.cpp [hunk 1]
```diff
@@ -394,28 +402,67 @@ llvm::Error ONNXModelLoader::loadSlice(const ONNX_NAMESPACE::NodeProto &op,
...
+  std::vector<ssize_t> starts;
+  std::vector<ssize_t> ends;
+  // Attribute 'axes' is optional.
+  std::vector<ssize_t> axes;
+  if (this->opsetVersion_ >= 10) {
+    Constant *startsC = getConstantByNameOrNull(op.input(1));
+    Constant *endsC = getConstantByNameOrNull(op.input(2));
+
+    RETURN_ERR_IF_NOT(startsC, "Starts Tensor is not Constant.");
+    RETURN_ERR_IF_NOT(endsC, "Starts Tensor is not Constant.");
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp]: typo: `Starts -> Ends`

File: lib/Importer/ONNXModelLoader.cpp [hunk 2]
```diff
@@ -386,6 +386,14 @@ llvm::Error ONNXModelLoader::loadConstant(const ONNX_NAMESPACE::NodeProto &op,
   return llvm::Error::success();
 }
 
+template <typename T>
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp]: Please add Doxygen for this helper.

**Comment by ayermolo** [comment]
Done, thanks!

File: lib/Importer/ONNXModelLoader.cpp [hunk 3]
```diff
@@ -385,28 +403,67 @@ llvm::Error ONNXModelLoader::loadSlice(const ONNX_NAMESPACE::NodeProto &op,
...
+    if (endsC->getElementType() == ElemKind::Int64ITy) {
+      helperSetter<int64_t>(endsC, ends);
+    } else if (endsC->getElementType() == ElemKind::Int32ITy) {
+      helperSetter<int32_t>(endsC, ends);
+    } else {
+      RETURN_ERR_IF_NOT(false, "Ends Tensor has unsupported type.");
+    }
+
+    if (op.input_size() > 3) {
+      Constant *axesC = getConstantByNameOrNull(op.input(3));
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp]: Whoops just realized you also need to check if this is null and return an error, similar to above with `RETURN_ERR_IF_NOT(startsC, "Starts Tensor is not Constant.");`

Comment (ayermolo) [reply lib/Importer/ONNXModelLoader.cpp]: Good catch, thanks. I also added a check for Steps.

**Pull request closed without merging**